cryptoinvestor/models/user.py

```diff
@@ -25,15 +24,16 @@ def buy(self, count: float, a_name: str) -> float:
 
         s = count * price
 
-        if self.balance() < s:
+        if self.account.balance < s:
             raise self.Error(
-                f"Buying {count} of {a_name} would cost {s} {lc}. You have {self.balance()} {lc}"
+                f"Buying {count} of {a_name} would cost {s} {lc}."
+                f"You have {self.account.balance} {lc}"
             )
 
-        owned = self.bought.get(a_name, 0)
+        owned = self.account.investments.get(a_name, {}).get('owned', 0)
 
-        self.bought[a_name] = owned + count
-        self._buy(s)
+        self.account.investments[a_name] = {'owned': (owned + count), 'price': price}
```

Comment:
What if the user buys the same currency at different points in time? (different currency prices)

Reply:
History of those prices could be implemented, sth in a way `..., 'price': [price1, price2, ...]`. Any other suggested ideas ?

Reply:
I'd do something like

```
a_name: [{
  'price': price1,
  'amount': amount1
}, {
  'price': price2,
  'amount': amount2
}]
```

We don't need to have the 'owned' layer as the whole structure is in `investments`

Reply:
This seems more like a history of purchases, the original intention was to display just how much of a given currency we have in our account, wasn't it ?

Reply:
Yes, but for computing how much the user has gained correctly it is required to have both at what cost he bought the currency and how much of the currency he has bought

Reply:
These would probably be displayed as different investments in the listview

Reply:
How do we subtract when selling then ? We sum the amounts in our investments to check if we have enough to sell and then we add `'amount'` with negative value ? Or should we have two dictionaries one for buy and the other for sell ?

Reply:
The first option seems reasonable

Reply:
Current Value seems to be redundant now. 
<img width="1076" alt="screen shot 2018-05-26 at 4 20 25 pm" src="https://user-images.githubusercontent.com/9095130/40577254-eb365770-6102-11e8-8fbe-c9b1dca301ff.png">
Suggested graphics are shown below, where profit shows difference in purchase value and its current value in percentage. Meaning if user sold at low price, and meanwhile it increased it is shown as loss with red text and with green text as gain if price dropped even more after being sold. (Data displayed is just to see the implementation, might not be accurate)
<img width="1066" alt="screen shot 2018-05-26 at 4 35 38 pm" src="https://user-images.githubusercontent.com/9095130/40577255-eb4ef67c-6102-11e8-9ded-a3256c36f23f.png">

Reply:
That looks awesome 👍